lib/utils.js

```diff
@@ -68,45 +43,34 @@ const getDataStats = (data, checksum, dedupHash) => ({
   size: Buffer.byteLength(data),
 });
 
-const compress = (file, minCompressSize, compression, cb) => {
-  fs.stat(file, (err, stats) => {
-    if (err || stats.size <= minCompressSize) {
-      cb(err, false);
-      return;
-    }
-    const filec = file + 'z';
-    const compressor = compressors[compression];
-    if (!compressor) {
-      throw new Error(`Unknown compression type ${compression} specified`);
-    }
-    compressor
-      .compressFile(file, filec)
-      .then(() =>
-        fs.rename(filec, file, err => {
-          if (err) cb(err, false);
-          else cb(null, true);
-        })
-      )
-      .catch(err => cb(err, false));
-  });
+const compress = async (file, minCompressSize, compression) => {
+  const compressor = compressors[compression];
+  if (!compressor) {
+    throw new Error(`Unknown compression type ${compression} specified`);
+  }
+
+  const stats = await fs.stat(file);
+  if (stats.size <= minCompressSize) return false;
+
+  const filec = file + 'z';
+  await compressor.compressFile(file, filec);
+  await fs.rename(filec, file);
+  return true;
 };
 
-const uncompress = (file, opts, cb) => {
-  fs.access(file, err => {
-    if (err) {
-      cb(err);
-      return;
-    }
-    const compressor = compressors[opts.compression];
-    if (!compressor) {
-      throw new Error(`Unknown compression type ${opts.compression} specified`);
-    }
+const uncompress = async (file, opts) => {
+  const compressor = compressors[opts.compression];
+  if (!compressor) {
+    throw new Error(`Unknown compression type ${opts.compression} specified`);
+  }
+
+  return new Promise((res, rej) => {
```

Comment:
The naming is not very obvious here:
```suggestion
  return new Promise((resolve, reject) => {
```